include/ensmallen_bits/pso/pso.hpp

```diff
@@ -0,0 +1,144 @@
+/**
+ * @file pso.hpp
+ * @author Chintan Soni
+ * @author Suryoday Basak
+ *
+ * Particle swarm optimization.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_PSO_PSO_HPP
+#define ENSMALLEN_PSO_PSO_HPP
+
+#include "update_policies/lbest_update.hpp"
+#include "init_policies/default_init.hpp"
+#include "init_policies/feasible_init.hpp"
+
+namespace ens {
+
+/**
+ * Particle swarm optimization (PSO) is an evolutionary optimization approach.
+ * PSO can optimize functions continuous as well as discrete functions. It does
+ * not require the gradient to be computable for its working. For PSO to work,
+ * the ArbitraryFunctionType parameter is required.
+ *
+ * This class must implement the following function:
+ *
+ *   double Evaluate(const arma::mat& coordinates);
+ *
+ * For more information, refer to the following paper:
+ *
+ * @inproceedings{Kennedy,
+ * doi = {10.1109/icnn.1995.488968},
+ * url = {https://doi.org/10.1109/icnn.1995.488968},
+ * publisher = {{IEEE}},
+ * author = {J. Kennedy and R. Eberhart},
+ * title = {Particle swarm optimization},
+ * booktitle = {ICNN}{\textquotesingle}95 - 
+ * 	International Conference on Neural Networks}
+ *}
+ */
+template<typename VelocityUpdatePolicy = LBestUpdate,
+         typename InitPolicy = DefaultInit>
+class PSOType
+{
+ public:
+  /**
+   * Construct the particle swarm optimizer with the given function and
+   * parameters. The defaults here are not necessarily good for the given
+   * problem, so it is suggested that the values used be tailored to the task
+   * at hand.
+   *
+   * @param numParticles Number of particles in the swarm.
+   * @param maxIterations Number of iterations allowed.
+   * @param exploitationFactor Influence of the personal best of the particle.
+   * @param explorationFactor Influence of the neighbours of the particle.
+   * @param VelocityUpdatePolicy Determines how the velocity of the particles are iteratively updated.
+   * @param InitPolicy Determines how the particles are initialized.
+   */
+  PSOType(
```

Comment:
Picky style comment, can we start right after the `(`; move the parameters one line up.